scripts/microupload.py

```diff
@@ -64,22 +127,40 @@ def main(args: List[str]) -> None:
                      for x in list_files(root, opts['--exclude'])]
     else:
         to_upload = [rel_root]
-
     created_cache = set()
     for path in progress('Uploading files', to_upload):
         local_path = os.path.abspath(path)
-        remote_path = os.path.normpath(path)
+        remote_path = os.path.normpath(path).replace(os.path.sep, '/')
         if verbose:
             print('\n{} -> {}'.format(local_path, remote_path),
                   file=sys.stderr, flush=True)
         remote_dir = os.path.dirname(path)
         if remote_dir:
             make_dirs(files, remote_dir, created_cache)
+            
         with open(local_path, 'rb') as fd:
-            files.put(remote_path, fd.read())
+            if different:
+                raw = fd.read()
+                local_file_size = str(os.stat(local_path)[6])
+                local_file_hash = sha1(raw).hexdigest()
+
+                remote_file_size = files.get_size(remote_path)
+                if local_file_size == remote_file_size:
+                    remote_file_hash = files.get_hash(remote_path)
+                    if remote_file_hash == local_file_hash:
+                        print("File Identical... Skipping upload", file=sys.stderr, flush=True)
+                    else:
+                        wait_for_board()
+                        files.put(remote_path, raw)
+                else:
+                    wait_for_board()
+                    files.put(remote_path, raw)
```

Comment:
You can simplify to something like,
```suggestion
                if local_file_size == files.get_size(remote_path) and local_file_hash == files.get_hash(remote_path):
                    print("File Identical... Skipping upload", file=sys.stderr, flush=True)
                else:
                    wait_for_board()
                    files.put(remote_path, raw)
            else:
                files.put(remote_path, fd.read())
```

The interpreter will short-circuit if the size check `local_file_size == files.get_size(remote_path)` evaluates false, and not calculate the hash too.